integration/oasis/stationgraph/node.go

```diff
@@ -0,0 +1,45 @@
+package stationgraph
+
+import (
+	"math"
+
+	"github.com/golang/glog"
+)
+
+type chargeInfo struct {
+	arrivalEnergy float64
+	chargeTime    float64
+	chargeEnergy  float64
+}
+
+type node struct {
+	id    nodeID
+	edges []*edge
```

Comment:
`edges` should not belong to `node`. Better to represent the relations like from a `node` via an `edge` to a `node` in `graph`, that's part of graph's adjacency list. Here's simple [example](https://github.com/wangyoucao577/algorithms_practice/blob/f45534d538b95f9f792940631c0e86f6b3bf6fa7/graph/adjacency_list_graph.go#L11).

Reply:
Here I try to build a node based graph, graph contains `nodes`,  and `nodes'` relation are represented as `edge`.   The graph's `orig` and `dest` node always be specified by sub-classes when create the graph.

Reply:
I see. By this design, `node` may have many neighbor `node`s, for each neighbor could have `weight/cost` for travelling to it. I'll suggest to use `neigbhor` instead of `edge` to name it. Also, it's better to describe the idea into package/type comments, especially for the `node/edge/graph/neighbor`.

Reply:
`neighbor` is good, will create `doc.go` file and records those information.